### PR TITLE
OpenVX extensions - add support for MIGraphX node

### PR DESCRIPTION
**Pull request by asalmanp**

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
> I was able to successfully build and run the test in this PR.

@asalmanp @kiritigowda Same here. Can be merged

File: amd_openvx_extensions/amd_migraphx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    list(APPEND SOURCES
+            src/kernels.cpp
+            src/migraphx_node.cpp
+        )
+
+    if (HIP_FOUND AND migraphx_FOUND)
+        message("-- ${Green}${PROJECT_NAME} -- Building with HIP backend${ColourReset}")
+        set(ENABLE_HIP 1)
+        add_definitions(-DENABLE_HIP=${ENABLE_HIP} -D__HIP_PLATFORM_AMD__)
+        link_directories(${HIP_PATH}/lib)
```
Comment (rrawther) [inline amd_openvx_extensions/amd_migraphx/CMakeLists.txt:61]: probably we need to change to target_xxx definitions as per new CMake compliance

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        printf("%s", string);
+        if (string[len - 1] != '\n')
+            printf("\n");
+        fflush(stdout);
+    }
+}
+
+void read_input_digit(const int n, std::vector<float>& input_digit)
+{
+    std::ifstream file("../digits.txt");
```
Comment (rrawther) [inline tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:39]: please check if this relative path is correct. Looks like digits.txt is in the same folder

**Review by rrawther — CHANGES_REQUESTED** [review]
@asalmanp Please address review comments

File: tests/amd_migraphx_test/mnist/migraphx_node_test.cpp (hunk 2 above)
Comment (asalmanp) [reply tests/amd_migraphx_test/mnist/migraphx_node_test.cpp:39]: @rrawther the path is correct. The test executable is built in the build folder, and it's run from there. The digits is outside of the build folder.

File: amd_openvx_extensions/amd_migraphx/CMakeLists.txt (hunk 1 above)
Comment (asalmanp) [reply amd_openvx_extensions/amd_migraphx/CMakeLists.txt:61]: @rrawther as discussed, we'll change the CMake for all projects in a separate PR.

**Comment by LakshmiKumar23** [comment]
@kiritigowda Can you merge this if it is ready?

**Review by rrawther — APPROVED** [review]

**Review by LakshmiKumar23 — APPROVED** [review]

**Pull request merged**